.github/scripts/strategy-matrix/generate.py

```diff
@@ -0,0 +1,141 @@
+                        skip = False
+                if skip:
+                    continue
+
+            # RHEL:
+            # - 9.4 using GCC 12: Debug and Unity on linux/amd64.
+            # - 9.6 using Clang: Release and no Unity on linux/amd64.
+            if os['distro_name'] == 'rhel':
+                skip = True
+                if os['distro_version'] == '9.4':
+                    if f'{os['compiler_name']}-{os['compiler_version']}' == 'gcc-12' and build_type == 'Debug' and '-Dunity=ON' in cmake_args and architecture['platform'] == 'linux/amd64':
+                        skip = False
+                elif os['distro_version'] == '9.6':
+                    if f'{os['compiler_name']}-{os['compiler_version']}' == 'clang-any' and build_type == 'Release' and '-Dunity=OFF' in cmake_args and architecture['platform'] == 'linux/amd64':
+                        skip = False
+                if skip:
+                    continue
+
+            # Ubuntu:
+            # - Jammy using GCC 12: Debug and no Unity on linux/arm64.
+            # - Noble using GCC 14: Release and Unity on linux/amd64.
+            # - Noble using Clang 18: Debug and no Unity on linux/amd64.
+            # - Noble using Clang 19: Release and Unity on linux/arm64.
+            if os['distro_name'] == 'ubuntu':
+                skip = True
+                if os['distro_version'] == 'jammy':
+                    if f'{os['compiler_name']}-{os['compiler_version']}' == 'gcc-12' and build_type == 'Debug' and '-Dunity=OFF' in cmake_args and architecture['platform'] == 'linux/arm64':
+                        skip = False
+                elif os['distro_version'] == 'noble':
+                    if f'{os['compiler_name']}-{os['compiler_version']}' == 'gcc-14' and build_type == 'Release' and '-Dunity=ON' in cmake_args and architecture['platform'] == 'linux/amd64':
+                        skip = False
+                    if f'{os['compiler_name']}-{os['compiler_version']}' == 'clang-18' and build_type == 'Debug' and '-Dunity=OFF' in cmake_args and architecture['platform'] == 'linux/amd64':
+                        skip = False
+                    if f'{os['compiler_name']}-{os['compiler_version']}' == 'clang-19' and build_type == 'Release' and '-Dunity=ON' in cmake_args and architecture['platform'] == 'linux/arm64':
+                        skip = False
+                if skip:
+                    continue
+
+            # MacOS:
+            # - Debug and no Unity on macos/arm64.
+            if os['distro_name'] == 'macos' and not (build_type == 'Debug' and '-Dunity=OFF' in cmake_args and architecture['platform'] == 'macos/arm64'):
+                continue
+
+            # Windows:
+            # - Release and Unity on windows/amd64.
+            if os['distro_name'] == 'windows' and not (build_type == 'Release' and '-Dunity=ON' in cmake_args and architecture['platform'] == 'windows/amd64'):
+                continue
+
+
```

Comment:
I would add `else:` section here, so we do not run both unity and non-unity builds for every configuration; this means for most cases (macos, windows, selection of Linux configs) skip non-unity builds, and for the opposite selection of Linux configs, skip unity builds.